scripting/get5/stats.sp

```diff
@@ -227,8 +227,26 @@ public Action Stats_PlayerDeathEvent(Event event, const char[] name, bool dontBr
       IncrementPlayerStat(attacker, STAT_KILLS);
       g_PlayerRoundKillOrAssistOrTradedDeath[attacker] = true;
 
-      if (headshot)
+      if (headshot) {
         IncrementPlayerStat(attacker, STAT_HEADSHOT_KILLS);
+      } else {
+
+        // Knife cannot headshot, so there's no reason to check for knife kill on these.
+        // We need the weapon ID to reliably translate to a knife. The regular "bayonet" - as the only knife -  is not
+        // prefixed with "knife" for whatever reason, so searching weapon name strings is unsafe.
+        CSWeaponID weaponId = CS_AliasToWeaponID(weapon);
+
+        // Other than these constants, all knives can be found after CSWeapon_MAX_WEAPONS_NO_KNIFES.
+        // See https://sourcemod.dev/#/cstrike/enumeration.CSWeaponID
+        if (weaponId == CSWeapon_KNIFE
+        || weaponId == CSWeapon_KNIFE_GG
+        || weaponId == CSWeapon_KNIFE_T
+        || weaponId == CSWeapon_KNIFE_GHOST
+        || weaponId > CSWeapon_MAX_WEAPONS_NO_KNIFES) {
+          IncrementPlayerStat(attacker, STAT_KNIFE_KILLS);
+        }
```

Comment:
I had to change to this method instead of StrContains("knife"). As mentioned in the comments here, there's one knife that does not; the regular bayonet. This method should also be future-proof.